r/src/array_to_vector.cpp

```diff
@@ -920,7 +921,7 @@ class Converter_Decimal : public Converter {
   Status Ingest_some_nulls(SEXP data, const std::shared_ptr<arrow::Array>& array,
                            R_xlen_t start, R_xlen_t n, size_t chunk_index) const {
     auto p_data = REAL(data) + start;
-    const auto& decimals_arr = checked_cast<const arrow::Decimal128Array&>(*array);
+    const auto& decimals_arr = checked_cast<const arrow::DecimalArray&>(*array);
```

Comment:
```suggestion
    const auto& decimals_arr = checked_cast<const DecimalArray&>(*array);
```

Reply:
so that it picks up the `DecimalArray` that was defined with the `using DecimalArray = ...`

Reply:
Done!